crates/rune/src/fmt/comments.rs

```diff
@@ -106,12 +105,11 @@ fn parse_line_comment(chars: &mut Peekable<impl Iterator<Item = (usize, char)>>)
     last_i + 1
 }
 
-fn parse_block_comment(chars: &mut Peekable<impl Iterator<Item = (usize, char)>>) -> Option<usize> {
+fn parse_block_comment(chars: &mut CharIndices<'_>) -> Option<usize> {
     while let Some((_, c)) = chars.next() {
         if c == '*' {
-            if let Some((_, '/')) = chars.peek() {
-                let (offset, _) = chars.next()?;
-                return Some(offset);
+            if let Some((_, '/')) = chars.clone().next() {
```

Comment:
This pattern is actually kinda interesting: For character iterators peekable is pretty much a waste. The char iterator itself is three pointers big - a slice iterator (two pointers) and the index (one usize). Cloning it to peek is actually fine.